tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.md

```diff
@@ -10,17 +10,108 @@ To verify vic-machine-server can create a VCH with a specified configuration
 This test requires a vSphere system where VCHs can be deployed
 
-# Test Steps:
-1. Create a VCH with as minimal a configuration as possible
-2. Inspect that VCH using the CLI
-3. Verify that the VCH appliance created has been successfully initialized
-4. Create a VCH with a more complex configuration (some input params for creation have placeholder values)
-5. Inspect that VCH using the CLI
-
-# Expected Outcome:
-* The results of 2 should contain the same information as was supplied when the VCH was created in 1.
-* The results of 3 should show that the VCH appliance is initialized and docker endpoint is able to connect.
-* The results of 5 should contain the same information as was supplied when the VCH was created in 4.
-
-# Possible Problems:
-None known
+**Note:** Multiple versions of tests must be implemented:
+ * With and without datacenter-scoped URLs.
+ * Using both username/password- and session-based authentication.
+
+It should not be necessary to implement all four combinations for every test case, but full coverage should be provided for at least one of the basic test cases.
+
+
+Basic Validation
+----------------
+
+These tests are intended to verify the feature at a basic level.
+
+###  1. Create a simple VCH (i.e., one using as many defaults as possible)
+
+###  2. Create a complex VCH (i.e., one which explicitly specifies as many settings as possible)
+
+
+Negative Cases
+--------------
+
+These tests are intended to verify the behavior in various failure cases. We must gracefully handle invalid input and unexpected user behavior.
+
+###  1. Attempt to create a VCH with invalid operations credentials
+
+###  2. Attempt to create a VCH with an invalid datastore
+
+###  3. Attempt to create a VCH with an invalid storage settings
+
+###  4. Attempt to create a VCH with an invalid networking settings
+
+###  5. Attempt to create a VCH with an invalid gateway settings
+
+###  6. Attempt to create a VCH with a name containing invalid characters
+
+(Not implemented.)
+
+###  7. Attempt to create a VCH with a name that is already in use
+
+(Not implemented.)
+
+###  8. Attempt to create a VCH with an invalid container name convention
+
+(Not implemented.)
+
+###  9. Attempt to create a VCH specifying an ID
+
+(Not implemented.)
+
+### 10. Attempt to create various VCHs with invalid resource settings
+
+(Not implemented.)
+
+### 11. Attempt to create various VCHs with invalid registry settings
+
+(Not implemented.)
+
+### 12. Attempt to create various VCHs with invalid security settings
+
+(Not implemented.)
+
+
+Interoperability
+----------------
+
+These tests are intended to verify that the API and CLI can coexist without issue.
+
+###  1. Verify that the CLI can be used to delete a VCH created by the API
+
+(Not implemented.)
+
+
+Concurrency
+-----------
+
+These tests are intended to verify that the API behaves as expected when performing concurrent operations.
+
+###  1. Attempt to create two VCHs at the same time
```

Comment:
Is it a good idea to test this in some separate cases like the following?
1. Attempt to create two VCHs within the same datacenter
2. Attempt to create two VCHs within different datacenter
I'm not sure if the scenario of same/different datacenter is necessary to test here, but just a thought. (also I'm not sure if the VCH name uniqueness is in datacenter level or cluster level)